navigator-menu/src/components/github-components/theme-toggle/ThemeToggle.jsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Wrapper,
+  ToggleBtns,
+  H2,
+  ToggleTextDark,
+  ToggleTextLight,
+} from './ThemeToggle.styles';
+import { useState } from 'react';
+
+function Toggle() {
+  const [btnToggle, setBtnToggle] = useState(false);
+
+  const toggle = (e) => {
+    let currentId = e.target.id;
+
+    if (currentId.includes('dark')) {
+      let htmlDoc = document.getElementsByTagName('html')[0];
+      htmlDoc.classList.add('dark');
+
+      setBtnToggle(true);
+    } else if (currentId.includes('light')) {
+      let htmlDoc = document.getElementsByTagName('html')[0];
+      htmlDoc.classList.remove('dark');
+
+      setBtnToggle(false);
+    }
+  };
+
+  const darkHoverEnter = () => {
+    let text = document.getElementById('dark-text');
+    text.classList.add('text-[#212630]');
+    let icon = document.getElementById('dark-icon');
+    icon.classList.remove('fill-[#697C9A]');
+    icon.classList.add('fill-[#212630]');
+  };
+
+  const darkHoverLeave = () => {
+    let text = document.getElementById('dark-text');
+    text.classList.remove('text-[#212630]');
+    let icon = document.getElementById('dark-icon');
+    icon.classList.remove('fill-[#212630]');
+    icon.classList.add('fill-[#697C9A]');
+  };
+
+  const lightHoverEnter = () => {
+    let text = document.getElementById('light-text');
+    text.classList.add('text-[#90a4d4]');
+    let icon = document.getElementById('light-icon');
+    icon.classList.remove('fill-[#697C9A]');
+    icon.classList.add('fill-[#90a4d4]');
+  };
+
+  const lightHoverLeave = () => {
+    let text = document.getElementById('light-text');
+    text.classList.remove('text-[#90a4d4]');
+    let icon = document.getElementById('light-icon');
+    icon.classList.remove('fill-[#90a4d4]');
+    icon.classList.add('fill-genericWhite');
+  };
+
+  return (
+    <Wrapper>
+      <H2>devfinder</H2>
+      <ToggleBtns>
```

Comment:
You have two components that have 1 or 2 lines difference. Think on how to combine them.